src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodOrUserDefinedOperatorSymbol.cs

```diff
@@ -69,12 +69,6 @@ public sealed override bool ReturnsVoid
                 }
             }
 
-            if (!IsPartial)
-            {
-                LazyAsyncMethodChecks(CancellationToken.None);
-                Debug.Assert(state.HasComplete(CompletionPart.FinishAsyncMethodChecks));
-            }
```

Comment:
Was this incorrect?

Reply:
> Was this incorrect?

It was done too early. Apparently, for no good reason. Was causing a cycle because the check needs to access constraints, which this method is responsible to figure out in case of overriding.